apisix/plugins/auth-casdoor.lua

```diff
@@ -0,0 +1,138 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local core = require("apisix.core")
+local cjson = require("cjson")
+local http = require("resty.http")
+local session = require("resty.session")
+local string = require("string")
+local log = core.log
+local ngx = ngx
+
+local plugin_name = "auth-casdoor"
+local schema = {
+    type = "object",
+    properties = {
+        casdoor_endpoint = {type = "string"},
+        client_id = {type = "string"},
+        client_secret = {type = "string"},
+        callback_url = {type = "string"},
+        test = {type = "boolean", default = false}
+    },
+    required = {
+        "callback_url", "casdoor_endpoint", "client_id", "client_secret"
+    }
+}
+
+local _M = {version = 0.1, priority = 2559, name = plugin_name, schema = schema}
+
+local function get_path(uri)
```

Comment:
We can use
https://github.com/apache/apisix/blob/acf7a0cc7f05aa15de21eb1aac798566f45bcb66/apisix/plugins/http-logger.lua#L88
to get the path instead of writing own solution.